.github/workflows/build_apk.yml

```diff
@@ -1,54 +1,76 @@
 name: Build prebuilt APK
 
 on:
-  workflow_dispatch:
   push:
-    branches:
+    branches: 
       - 'A12.1'
+    paths-ignore:
+      - '.github/**'
+      - '**.mkdn'
+      - '.**'
+  pull_request:
+      paths-ignore:
+      - '.github/**'
+      - '**.mkdn'
+      - '.**'
+  workflow_dispatch:
 
 jobs:
   build-release-apk:
     runs-on: ubuntu-latest
     continue-on-error: true
+    if: ${{ !startsWith(github.event.head_commit.message, '[skip]') }}
+    
     steps:
       - name: Check out repository
         uses: actions/checkout@v3
+
       - name: Clone our prebuilts
         run: |
           git clone https://.:${{ secrets.API_TOKEN_GITHUB }}@github.com/Flamingo-OS/vendor_prebuilts
           mkdir app/libs
           cp vendor_prebuilts/libs/framework.jar vendor_prebuilts/libs/support-compose.aar app/libs/
+
       - name: Set up Java 11
         uses: actions/setup-java@v3
         with:
           distribution: 'zulu'
           java-version: 11
-      - uses: gradle/gradle-build-action@v2
-      - name: Grant execution permission to Gradle Wrapper
-        run: chmod +x gradlew
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
       - name: Build release APK
         run: ./gradlew assembleRelease
+
       - name: Rename artifact
+        if: ${{ !github.event.pull_request }}
```

Comment:
it should work only for commit. `${{ github.event.commit }}` if i am correct. It shouldn't be just for not pull request